src/main/java/com/getindata/connectors/http/internal/table/lookup/querycreators/GenericJsonQueryCreatorFactory.java

```diff
@@ -47,8 +48,8 @@ public LookupQueryCreator createLookupQueryCreator(
             queryFormatAwareConfiguration
         );
 
-        SerializationSchema<RowData> serializationSchema =
-            encoder.createRuntimeEncoder(null, lookupRow.getLookupPhysicalRowDataType());
+        SerializationSchema<RowData> serializationSchema = new SynchronizedSerializationSchema<>(
```

Comment:
Maybe we could wrap with synchronized schema only when async is used? That way we could avoid unnecessary lock creation for sync connector calls where we should be thread safe anyway

Reply:
Thanks! Fixed.